presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportServlet.java

```diff
@@ -116,7 +116,44 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
     protected void reportFailure(HttpServletResponse response, String message)
             throws IOException
     {
-        response.sendError(SC_BAD_REQUEST, message);
+        response.sendError(SC_BAD_REQUEST, sanitizeMessage(message));
+    }
+    private String sanitizeMessage(String message)
+    {
+        if (message == null) {
+            return "";
+        }
+        StringBuilder stringBuilder = new StringBuilder(message.length());
+
+        // Traverse each character in the string and replace only when necessary
+        for (int i = 0; i < message.length(); i++) {
```

Comment:
Please put this in a utility class and add testing for this utility class.

Reply:
The comments have been addressed.